### PR TITLE
Helm values for gateway gzip options

### PR DESCRIPTION
**Pull request by kevin-shelaga**
# Description

add gateway gzip options to helm chart

# Context

User request: https://github.com/solo-io/gloo/issues/4405

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/4405

### CODE REVIEW

**Comment by kevin-shelaga** [comment]
Doc / Generate versioned docs site check is failing with the following

```sh
Error: Must either set GITHUB_TOKEN or set SKIP_CHANGELOG_GENERATION environment variable to true
Usage:
  docs-util gen-changelog-md [flags]

Flags:
  -h, --help   help for gen-changelog-md

Global Flags:
      --call-latest      if set, will use the string 'latest' in the scope, rather than the particular release version
      --no-scope         if set, will not nest the served docs by product or version
      --product string   product to which the docs refer (defaults to gloo) (default "gloo")
      --version string   version of docs and code

exit status 1
make[1]: *** [site-common] Error 1
Error: Makefile:21: recipe for target 'site-common' failed
make[1]: Leaving directory '/home/runner/work/gloo/gloo/docs/gloo-temp/docs'
make: *** [build-site] Error 2
Error: Makefile:74: recipe for target 'build-site' failed
make: Leaving directory '/home/runner/work/gloo/gloo/docs'
Error: Process completed with exit code 2.
```

**Comment by kevin-shelaga** [comment]
Docs were generated, not sure what to do to fix [Docs / Generate versioned docs site (pull_request)](https://github.com/solo-io/gloo/pull/4424/checks?check_run_id=2102217919)

File: install/helm/gloo/generate/values.go [hunk 1]
```diff
@@ -315,12 +316,14 @@ type GatewayProxyGatewaySettings struct {
 	CustomHttpsGateway       string                   `json:"customHttpsGateway,omitempty" desc:"custom yaml to use for https gateway settings"`
 	GatewayOptions           v1.GatewayOptions        `json:"options,omitempty" desc:"custom options for http(s) gateways"`
 	AccessLoggingService     als.AccessLoggingService `json:"accessLoggingService,omitempty"`
+	Gzip                     gzip.Gzip                `json:"gzip,omitempty"`
```
Comment (kdorosh) [inline install/helm/gloo/generate/values.go:319]: let's add a description. also looks like we need to run codegen again

File: install/helm/gloo/templates/8-default-gateways.yaml [hunk 2]
```diff
@@ -21,9 +21,14 @@ spec:
   bindAddress: "::"
   {{- end }}
   bindPort: {{ $spec.podTemplate.httpPort }}
-{{- if $gatewaySettings.customHttpGateway}}
+{{- if or (and ($gatewaySettings.customHttpsGateway) (not $gatewaySettings.gzip)) }}
   httpGateway:
-{{ toYaml $gatewaySettings.customHttpGateway | indent 4}}
+{{ toYaml $gatewaySettings.customHttpsGateway | indent 6}}
+{{- else if or (and ($gatewaySettings.gzip) (not $gatewaySettings.customHttpsGateway)) }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/8-default-gateways.yaml:27]: not a huge fan of this extra complexity, users can already use `customHttpGateway` to accomplish the same thing with roughly the same user experience. Not convinced this is something we want to support right now

Comment (kevin-shelaga) [reply install/helm/gloo/templates/8-default-gateways.yaml:27]: @kdorosh fair point, customHttpGateway can accomplish the same thing. I will go ahead and close this PR

**Pull request closed without merging**